### ISSUE TITLE
Add HOWTO: Save/Restore only the inference variables

### ISSUE THREAD

**Issue by marcvanzee** [google/flax#880]
For instance, one may want to drop the Adam and optimizer variables.

To save only the params:

```
checkpoints.save_checkpoint(model_dir, optimizer.target, step)
```

To restore only the params:

```
new_params = checkpoints.restore_checkpoint(model_dir, optimizer.target)
optimizer = optimizer.replace(target=new_params)
```

Note: the optimizer now has `optimizer.state.step = 0` after restoring the checkpoint, so you probably want to checkpoint `(step, optimizer.target)` to avoid resetting things like learning rate schedules.

**Comment by marcvanzee** [comment]
Restoring only part of a model is very simple as well:

```
params = {'encoder': initialized_params['encoder'], 'decoder': checkpoint_params['decoder']}
```

**Comment by marcvanzee** [comment]
Closed by #923
